### ISSUE TITLE
Bug: AssignmentPattern.left is not just BindingName

### ISSUE THREAD

**Issue by Josh-Cena** [typescript-eslint/typescript-eslint#6159]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Relevant Package

ast-spec

### Playground Link

https://typescript-eslint.io/play#ts=4.9.3&jsx=true&sourceType=module&showAST=es&code=GYVwdgxgLglg9mABMOcAUBvRBDRBfRAXkQzwBpEBtAIwF0iraBKEgKFb1aA&eslintrc=N4KABGBEBOCuA2BTAzpAXGYBfEWg&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA

### Repro Code

```typescript
function foo({ a } = {}, [b] = []) {

}
```

### Expected Result

When you check the AST it generates, it contains:

```
params:
  0: AssignmentPattern
    left: ObjectPattern
    right: ObjectExpression
  1: AssignmentPattern
    left: ArrayPattern
    right: ArrayExpression
```

### Actual Result

The `ast-spec` type declaration says:

https://github.com/typescript-eslint/typescript-eslint/blob/bed6356866c04e6e4d5db3f71f31583dbbe0f36c/packages/ast-spec/src/parameter/AssignmentPattern/spec.ts#L8-L15

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/parser`            | `5.45.0` |
| `@typescript-eslint/typescript-estree` | `5.45.0` |

**Comment by bradzacher** [comment]
The types are correct here - `BindingName` includes `BindingPattern` which includes `ObjectPattern | ArrayPattern`

https://github.com/typescript-eslint/typescript-eslint/blob/bed6356866c04e6e4d5db3f71f31583dbbe0f36c/packages/ast-spec/src/unions/BindingName.ts#L4

https://github.com/typescript-eslint/typescript-eslint/blob/bed6356866c04e6e4d5db3f71f31583dbbe0f36c/packages/ast-spec/src/unions/BindingPattern.ts#L4

**Comment by Josh-Cena** [comment]
Ah, my mind is failing me 😢
